data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessor.java

```diff
@@ -215,6 +233,40 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
         }
         return resultRecords;
     }
+    
+    /**
+     * Waits for the circuit breaker to close before proceeding.
+     * This method will block execution until the circuit breaker is closed.
+     */
+    private void waitForCircuitBreakerToClosed() {
+        if (!circuitBreaker.isPresent()) {
+            return;
+        }
+        
+        CircuitBreaker breaker = circuitBreaker.get();
+        if (breaker.isOpen()) {
+            LOG.warn("Circuit breaker is open. Pausing Lambda invocation to prevent OOM.");
+            circuitBreakerTripsCounter.increment();
+            
+            // Wait until the circuit breaker is closed
+            boolean isFirstIteration = true;
+            while (breaker.isOpen()) {
```

Comment:
This should not be indefinite wait. We should wait for 10 or 15 seconds and come out even the circuit breaker is open.